widget/fileicon.go

```diff
@@ -73,7 +72,12 @@ func (i *FileIcon) CreateRenderer() fyne.WidgetRenderer {
 
 	i.setURI(i.URI)
 
-	s := &fileIconRenderer{file: i}
+	// TODO should FileIcon render a background representing selection, or should it be in the collection?
+	// TODO file dialog currently uses a container with NewGridWrapLayout, but if this changes to List, or Table then the primary color background would be rendered twice.
+	bg := canvas.NewRectangle(theme.PrimaryColor())
```

Comment:
Hmm, I am not so sure if I think the base fileicon should have a background at all. We probably need a better way to visually indicate the selection, I guess.

Reply:
Agreed, I added the background so the behaviour is consistent and it made more sense to add it here as FileIcon has SetSelected than to add it to the grid. Do you think `SetSelected` and the general selection code should be deprecated and removed from FileIcon? A lot depends on FileDialog. If we're agreed that the right approach is to use List & Table (would be nice to have a toggle switch so users can pick how they want the files laid out) to display the files/directories and the collections will have selection indication then the background can be removed from FileIcon. However we may still want the functionality to invert the colours, but if we do it for one widget we may have to do it for all.

Reply:
That sounds like a splendid idea. Switching to table and list would be a big win performance wise and then let them handle the selection indication. Yeah I can see colour inversion might need to be done for that.

Reply:
Note: I dont’t think this is limited to list or table but every container that allows selection or other indications should handle this by itself instead of delegating it to the widget.

Reply:
Agreed, tho it should be noted that changing the background to primary colour to indicate selection can sometimes make child canvasobject less readable, hence the colour invertion.

Perhaps the preferred approach should be the primary colour shadow "halo" effect? The downside with this is that the halo is rendered outside the bounds of the canvasobject.